### ISSUE TITLE
Group chat - App crashes when navigating to group details page after creating the group

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#45466]
- [ ] If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!

___

**Version Number:** 9.0.7-4
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go offline.
3. Go to FAB > Start chat > Chat.
4. Select a few users.
5. Create a group chat.
6. Click on the report header.
7. Click Name.
8. Go online.

## Expected Result:
App will not crash.

## Actual Result:
App crashes.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/e1eaed2f-6b4e-49d1-8b15-26a75b957deb

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
We think this issue might be related to the #vip-vsb

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App crashes.

### What is the root cause of that problem?
When we create a group chat, the `parentReportID` is an empty string and then after we go back online BE returns `parentReportID` of the report as `null`. The app crashes because the key is changed from collection key to collection member key

https://github.com/Expensify/App/blob/1465a9294f5ff357fb30f55c1e90df791c73cd29/src/pages/ReportDetailsPage.tsx#L90
### What changes do you think we should make in order to solve the problem?
Use `||` instead of `??` here

https://github.com/Expensify/App/blob/1465a9294f5ff357fb30f55c1e90df791c73cd29/src/pages/ReportDetailsPage.tsx#L89-L90

### What alternative solutions did you explore? (Optional)
Or when we create the optimistic data for the chat, add `parentReportID` field as `null` instead of empty string,

**Comment by puneetlath** [comment]
@nkdengineer do you know what PR introduced this issue?

**Comment by nkdengineer** [comment]
@puneetlath  I think it comes from https://github.com/Expensify/App/pull/45330.

**Comment by puneetlath** [comment]
Ok gotcha. @srikarparsi do you see any issues with @nkdengineer's proposed fix?

**Comment by srikarparsi** [comment]
I personally think the alternate solution is better. I think we should do that regardless of whether or not we implement the main solution. @nkdengineer is there a reason you prefer the main solution over the alternate, and is there a reason you would still want to implement the main solution on top of the alternate?

**Comment by chiragsalian** [comment]
Removing deployblocker label from this blocker since I'm pretty sure the issue here is related to web-E.

**Comment by puneetlath** [comment]
Ok let's go with @nkdengineer's alternate proposal. Can you raise a PR ASAP @nkdengineer?

**Comment by puneetlath** [comment]
I raised the PR but we will still compensate you @nkdengineer

**Comment by kavimuru** [comment]
@puneetlath able to reproduce

https://github.com/user-attachments/assets/95768cfa-232e-435b-af63-f077998bb82c

**Comment by srikarparsi** [comment]
I can take a look at this

**Comment by thienlnam** [comment]
Hmm, I tested this on staging before kicking off the deploy but I must have not followed the steps correctly - my bad, but it should be okay if we can get another fix CPed before the next deploy since this has to do with offline flows

**Comment by srikarparsi** [comment]
Yup, on it. It'll take me like 10-15 minutes probably if that's okay?

**Comment by nkdengineer** [comment]
> Ok let's go with @nkdengineer's alternate proposal. Can you raise a PR ASAP @nkdengineer?

It's too late in my time zone.

> I raised the PR but we will still compensate you @nkdengineer

@puneetlath Thanks. Can you please assign me here for payment?

**Comment by puneetlath** [comment]
@nkdengineer sent you an offer. Ping me here when you've accepted. https://www.upwork.com/nx/wm/offer/103157107

**Comment by nkdengineer** [comment]
@puneetlath I accepted, thanks 🙇

**Comment by puneetlath** [comment]
Paid thanks!
